### PR TITLE
Don't store the translated name of options / tool tips / values / page names.

### PR DESCRIPTION
**Pull request by BevapDin**
Translate them when needed. Move global variables into the `options_manager` class.

### CODE REVIEW

File: src/options.cpp [hunk 1]
```diff
@@ -484,12 +477,20 @@ std::string options_manager::cOpt::getValueName() const
...
+                if( elem.first == sDefault ) {
+                    return bTranslated ? _( elem.second.c_str() ) : elem.first;
+                }
+            }
+            return std::string();
+        } ();
         const std::string sItems = enumerate_as_string( vItems.begin(), vItems.end(),
-        [bTranslated ]( const std::string &elem ) {
-            return bTranslated ? optionNames[elem] : elem;
+        [bTranslated, &defaultName]( const std::pair<std::string, std::string> &elem ) {
```
Comment (codemime) [inline src/options.cpp]: `defaultName` is explicitly captured, but never used in the lambda.

File: src/options.cpp [hunk 2]
```diff
@@ -484,12 +477,20 @@ std::string options_manager::cOpt::getValueName() const
...
+            return bTranslated ? _( elem.second.c_str() ) : elem.first;
         }, false );
         return string_format(_("Default: %s - Values: %s"),
-                             (bTranslated) ? optionNames[sDefault].c_str() : sDefault.c_str(), sItems.c_str());
+                             defaultName.c_str(), sItems.c_str() );
```
Comment (codemime) [inline src/options.cpp]: Why is that a lambda after all? It could have been made a variable.

File: src/options.cpp [hunk 3]
```diff
@@ -484,12 +477,20 @@ std::string options_manager::cOpt::getValueName() const
 std::string options_manager::cOpt::getDefaultText(const bool bTranslated) const
 {
     if (sType == "string_select") {
+        const std::string defaultName = [bTranslated, this]() {
+            for( const auto &elem : vItems ) {
+                if( elem.first == sDefault ) {
```
Comment (codemime) [inline src/options.cpp]: Any particular reason not to use `std::find()`?

File: src/options.cpp [hunk 4]
```diff
@@ -745,780 +739,724 @@ static std::string build_resource_list(
...
     mOptionsSort["interface"]++;
 
-    optionNames["fahrenheit"] = _("Fahrenheit");
-    optionNames["celsius"] = _("Celsius");
-    add("USE_CELSIUS", "interface", _("Temperature units"),
-        _("Switch between Celsius and Fahrenheit."),
-        "fahrenheit,celsius", "fahrenheit"
+    add( "USE_CELSIUS", "interface", translate_marker( "Temperature units" ),
+        translate_marker( "Switch between Celsius and Fahrenheit." ),
+        { { "fahrenheit", translate_marker( "Fahrenheit" ) }, { "celsius", translate_marker( "Celsius" ) } }, "fahrenheit"
```
Comment (codemime) [inline src/options.cpp]: IMHO these blocks would benefit from more accurate formatting:
```c++
add( "USE_CELSIUS", "interface", 
  translate_marker( "Temperature units" ),
  translate_marker( "Switch between Celsius and Fahrenheit." ), {
    { "fahrenheit", translate_marker( "Fahrenheit" ) },
    { "celsius",    translate_marker( "Celsius" ) } 
  }, "fahrenheit"
);
```

File: src/options.cpp (hunk 2 above)
Comment (BevapDin) [inline src/options.cpp]: What lambda? `defaultName` is a `const std::string`.

File: src/options.cpp (hunk 4 above)
Comment (BevapDin) [inline src/options.cpp]: Sadly this formatting is not astyle-compatible. After applying astyle, I get
```C++

    add( "USE_CELSIUS", "interface",
         translate_marker( "Temperature units" ),
    translate_marker( "Switch between Celsius and Fahrenheit." ), {
        { "fahrenheit", translate_marker( "Fahrenheit" ) },
        { "celsius", translate_marker( "Celsius" ) }
    }, "fahrenheit"
       );
```

**Pull request merged**